docs/getting-started.md

```diff
@@ -256,3 +256,49 @@ import {mean, median} from "d3-array";
 ```
 
 TypeScript declarations are available via DefinitelyTyped.
+
+## D3 in React
+
+Most D3 modules (like d3-scale, d3-array, d3-color, d3-format, and d3-random) don’t interact with the DOM, so there is no difference when using them in React. You can use them in JSX to do purely declarative visualization, like this dot plot of one-dimensional numeric data:
+
+```jsx
+import {scaleLinear, extent} from "d3";
+
+function DotPlot({data, width}) {
+  const x = scaleLinear(extent(data), [5, width - 5]).nice(true);
+  return (
+    <svg width={width} height="4">
+      {data.map((d, i) => (
+        <circle key={i} cx={x(d)} cy="2" r="2" />
+      ))}
+    </svg>
+  );
+}
+```
+
+Some D3 modules (d3-selection, d3-transition, d3-axis, d3-brush, d3-zoom) do manipulate the DOM, which competes with React’s management of the DOM. In those cases, you can attach a ref to an element and pass it to D3 in a useEffect hook. For example, to add an axis to the example above:
+
+```jsx
+import {scaleLinear, extent, select, axisBottom} from "d3";
+import {useRef, useEffect} from "react";
+
+function DotPlot({data, width}) {
+  const x = scaleLinear(extent(data), [5, width - 5]).nice(true);
+  const ref = useRef();
+  useEffect(() => {
+    if (!ref.current) return;
+    select(ref.current).append("g")
+        .attr("transform", "translate(0, 4)")
+        .call(axisBottom(x).ticks(5));
+  }, [x]);
```

Comment:
This will append a new axis every time it renders, right? Because every time DotPlot renders, it’ll create a new `x` scale, triggering this affect. I think either you have to remove the axis you added in a disposal hook for the useEffect, or you have to use a join to replace the previous axis (if any) when the effect runs. 🤔